packages/hathor-rpc-handler/src/helpers/transactions.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet } from '@hathor/wallet-lib';
+import { HATHOR_TOKEN_CONFIG } from '@hathor/wallet-lib/lib/constants';
+import { QueryUtxosFilters, SendTxInput, SendTxInputSpecific, SendTxOutput, UtxoDetails, UtxoInfo } from '../types';
+import { NoUtxosAvailableError } from '../errors';
+
+export interface TokenOutput {
+  tokenUid: string;
+  amount: number;
+}
+
+export interface PreparedInput {
+  txId: string;
+  index: number;
+}
+
+export interface PreparedTxResponse {
+  success: boolean;
+  outputs: SendTxOutput[];
+  inputs: PreparedInput[];
+}
+
+export async function getUtxosToFillTx(
+  wallet: HathorWallet,
+  sumOutputs: number,
+  options: QueryUtxosFilters,
+): Promise<UtxoInfo[] | null> {
+  // We want to find only utxos to use in the tx, so we must filter by available only
+  const getUtxosOptions = {
+    ...options,
+    only_available_utxos: true
+  };
+  const utxosDetails: UtxoDetails = await wallet.getUtxos(getUtxosOptions);
+
+  // If we can't fill all the amount with the returned utxos, then return null
+  if (utxosDetails.total_amount_available < sumOutputs) {
+    return null;
+  }
+
+  const utxos: UtxoInfo[] = utxosDetails.utxos;
+
+  // Sort utxos with larger amounts first
+  utxos.sort((a, b) => b.amount - a.amount);
+
+  if (utxos[0].amount > sumOutputs) {
+    // If we have a single utxo capable of providing the full amount
+    // then we find the smallest utxos that can fill the full amount
+
+    // This is the index of the first utxo that does not fill the full amount
+    // if this is -1, then all utxos fill the full amount and we should get the last one
+    const firstSmallerIndex = utxos.findIndex(obj => obj.amount < sumOutputs);
+
+    if (firstSmallerIndex === -1) {
+      // Return the last element of the array (the one with smaller amount)
+      return [
+        utxos[utxos.length -1],
+      ];
+    }
+
+    // Return the element right before the first element that does not provide the full amount
+    return [utxos[firstSmallerIndex - 1]];
+  }
+
+  // Else we get the utxos in order until the full amount is filled
+  let total = 0;
+  const retUtxos = [];
+
+  for (const utxo of utxos) {
+    retUtxos.push(utxo);
+    total += utxo.amount;
+
+    if (total >= sumOutputs) {
+      return retUtxos;
+    }
+  }
+
+  return null;
+}
+
+export async function prepareTxFunds(
```

Comment:
Missing docstring

We probably have a method in the lib that does exactly this but after understanding what you want with the method after the docstring is added, then we can see it.